lib/segment/src/data_types/index.rs

```diff
@@ -210,6 +210,101 @@ pub struct TextIndexParams {
     /// If true, store the index on disk. Default: false.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub on_disk: Option<bool>,
+
+    /// Algorithm for stemming. Default: disabled.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub stemmer: Option<StemmingAlgorithm>,
+}
+
+/// Different stemming algorithms with their configs.
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Clone, PartialEq, Hash, Eq)]
+#[serde(tag = "type", rename_all = "snake_case")]
```

Comment:
Using serde tag will generate anonymous object, which is very hard to use in python client. Istead,